PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/projects/HDInsightsScalaProjectWizard.java

```diff
@@ -134,11 +141,23 @@ public boolean canFinish() {
 	@Override
 	public boolean performFinish() {
 		try {
-			CreateProjectUtil.createSampleFile(this.id, this.hdInsightScalaPageOne.getProjectName());
+			CreateProjectUtil.createSampleFile(this.id, this.hdInsightScalaPageOne.getProjectName(), this.useMaven, this.sparkVersion);
 		} catch (CoreException e) {
 			Activator.getDefault().log("Create HDInsight project error", e);
 		}
+		
+		// Configure Java project first and then enable Maven nature, otherwise the classpath will be overwritten
+		boolean result = super.performFinish();
+		if (useMaven) {
+			try {
+				MavenPlugin.getProjectConfigurationManager().enableMavenNature(this.hdInsightScalaPageTwo.getJavaProject().getProject(), 
+						new ResolverConfiguration(), 
+						new NullProgressMonitor());
```

Comment:
>NullProgressMonitor() [](start = 10, length = 21)

It would be better if we have a ProgressMonitor here since resolving may need a long time

Reply:
Good point. Updated.

---
In reply to: [136775504](https://github.com/Microsoft/azure-tools-for-java/pull/984#discussion_r136775504) [](ancestors = 136775504)